### DISCUSSION TITLE
Missing Features?

### DISCUSSION THREAD

**Discussion by Karric** [microsoft/BotFramework-Composer#6695] [Q&A]
I'm new to MBF, mostly trying to stick with Composer. I get that anything is possible with custom C# classes, but I just want to confirm/deny if the following things are possible **within Composer**:

- ingest extra data from the DirectLine API request payload into the bot's memory.
- add extra properties to the DirectLine API response object.
- create custom C# classes/functions entirely within Composer. For example, a function that reverses the letters of a person's name.

**Comment by gary-x0pa** [comment]
First two is not supported i think. Last one is supported via https://docs.microsoft.com/en-us/composer/how-to-add-custom-action?tabs=csharp slightly crude approach but it works
